internal/receiver/discoveryreceiver/config.go

```diff
@@ -0,0 +1,162 @@
+// Copyright Splunk, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package discoveryreceiver
+
+import (
+	"fmt"
+
+	"go.opentelemetry.io/collector/config"
+	"go.uber.org/multierr"
+)
+
+var (
+	_ config.Receiver = (*Config)(nil)
+
+	allowedStatusTypes = []string{"successful", "partial", "failed"}
+	allowedStatuses    = func() map[string]struct{} {
+		sm := map[string]struct{}{}
+		for _, status := range allowedStatusTypes {
+			sm[status] = struct{}{}
+		}
+		return sm
+	}()
+
+	allowedMatchTypes = []string{"regexp", "strict", "expr"}
+)
+
+type Config struct {
+	// Receivers is a mapping of all receiver types to discovery and whose metrics
+	// and application statements to evaluate for assessing component status.
+	Receivers               map[config.ComponentID]ReceiverEntry `mapstructure:"receivers"`
+	config.ReceiverSettings `mapstructure:",squash"`
+	// The configured Observer extensions from which to receive Endpoint events.
+	// Must implement the observer.Observable interface.
+	WatchObservers []config.ComponentID `mapstructure:"watch_observers"`
+	// Whether to emit log records for all endpoint activity, consisting of Endpoint
+	// content as record attributes.
+	LogEndpoints bool `mapstructure:"log_endpoints"`
+	// Whether to include the receiver config as a base64-encoded "discovery.receiver.config"
+	// resource attribute string value. Will also contain the configured observer that
+	// produced the endpoint leading to receiver creation.
+	// Warning: these values will include the literal receiver subconfig from the parent Collector config
+	// and provides no secret redaction and is easily decodable into plaintext.
```

Comment:
Warning: these values will include the literal receiver subconfig from the parent Collector config. 
They provide no secret redaction, and are easily decodable into plaintext.